### PR TITLE
Add line numbers to Syntax Errors

### PR DESCRIPTION
**Pull request by ishaankolluri**
Adds a line count to scanner-level syntax errors.

### CODE REVIEW

**Review by Neitsch — CHANGES_REQUESTED** [review]
Please add a testcase WITH output to assert correctness.

**Comment by ishaankolluri** [comment]
To do:
* still have to increment the characters in the `read_string` buffer. Slightly more tricky. Will play around with it tomorrow.
* build a variety of test cases to illustrate the usage in comments, single line comments, with `extern`, etc.

**Comment by ishaankolluri** [comment]
Would love some feedback on correctness/best practice.

Also, would love some more ideas for interesting test cases here. Was thinking about testing `read_string`, but couldn't readily come up with an idea of how to break it.

**Comment by ishaankolluri** [comment]
Example:

``plt4115@plt4115 ~/Extend (parse_error) $ ./main.byte -c testcases/inputs/test_parse_error_comment.xtnd
Fatal error: exception Scanner.SyntaxError("Invalid character: $ on line 3 at character 30")``

File: src/main/parser.mly [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 %{
 open Ast
+
+(*let parse_error s =
+  print_endline s;
+  flush stdout*)
```
Comment (Neitsch) [inline src/main/parser.mly]: Remove comment.

**Review by Neitsch — APPROVED** [review]
Please add a testcase that uses a string. For example
foo := "Hello"; $$$;

File: src/main/parser.mly (hunk 1 above)
Comment (ishaankolluri) [reply src/main/parser.mly]: Sorry about this!

**Comment by oracleofnj** [comment]
cool!

**Pull request merged**